### ISSUE TITLE
Redraw randomly hangs until a keypress

### ISSUE THREAD

**Issue by valff** [alacritty/alacritty#824]
Using Alacritty (commit 6916537858d12e02478d618339d1b9f0e73a677a) on Ubuntu 16.04 with X11.

It is better described with the following screencast. Here I'm repeatedly opening and closing `tig`, so the output should be the same in each iteration. The screen often stays in a partially redrawed state (marked by mouse cursor movement). Then it finishes redraw correctly when I'm pressing a key.

![screencast](https://user-images.githubusercontent.com/12503689/31335101-cb888a5a-acf9-11e7-9e91-b6db6f6d4564.gif)

**Comment by Haggus** [comment]
I have the same exact bug. Ubuntu 16.04 with X11.

However, the interesting thing is this, bug appears only on when working in tmux.
Also, this bug doesn't exist on my Arch desktop (even when I'm using tmux).

**Comment by valff** [comment]
For me it is the same with and without tmux. I guess it can be hard to debug.

Please guide me how can I help with this, I'm familiar with rust.

**Comment by jwilm** [comment]
We've had a few isolated reports of this since the issue was originally "fixed". It's to the point I can no longer reproduce, but I am happy to advise you on doing the debugging if you're willing to get your hands dirty.

There are two primary threads in Alacritty: 1. the parser thread, and 2. the render/input thread. The parser thread reads from the pty asynchronously and locks the terminal struct once it's got some data. With lock in hand, the parser runs and updates what is visible on the screen. After the buffer is drained and terminal updated, this thread sends an event to the render/input thread which should trigger a wakeup. This all happens within [`EventLoop::pty_read`](https://github.com/jwilm/alacritty/blob/68cca6eb381f88e861439ed6a1dda1ab6a7429d3/src/event_loop.rs#L247-L301).

You might note that the render/input thread is only notified when the terminal `dirty` flag is not already set. As an initial sanity check, you might try disabling that check and always notify the render thread (this has a performance penalty which is why we don't already do it). At the same time, that might not actually be the problem and just change around timing enough to appear to solve the issue. Thus, this can be used to demonstrate that the dirty flag check is *not* the problem (would still have the issue), but it cannot conclusively conclude that it *is* the problem (issue might go away, but for different reason). In this latter case, you'll probably want to start adding some logging about when the render thread is kicked.

The part where this is consumed on the render thread starts in [_main.rs_](https://github.com/jwilm/alacritty/blob/68cca6eb381f88e861439ed6a1dda1ab6a7429d3/src/main.rs#L168-L201). The first line in this loop

```rust
let mut terminal = processor.process_events(&terminal, display.window());
```

will usually block and wait for events. Note that further down the loop, there's a check

```rust
if terminal.needs_draw() {
    // draw
}
```

So either this check would be returning false, or control never reaches this point.

Looking at [`Processor::process_events`](https://github.com/jwilm/alacritty/blob/master/src/event.rs#L336), this method is either blocking or polling for events based on initial state. Your draw problem ends up with it stuck blocking for events and never waking up, so we'll want to check that out. I would recommend adding some logging in that area. Finally, you might want to add some logging where the [`Awakened` event is handled](https://github.com/jwilm/alacritty/blob/68cca6eb381f88e861439ed6a1dda1ab6a7429d3/src/event.rs#L328).

I'm happy to help review the logs you generate or guide you through more of the code. Please let me know how else I can help; I would love to fully squash this bug, and I need help from someone like yourself to do it.

**Comment by valff** [comment]
@jwilm Thanks for a great guidance! Will find a spare time and start digging.

**Comment by koivunej** [comment]
I think I might be hit by the same issue and I have stayed on a rather old version of alacritty because of this (24c32dc). I don't think I can spare much effort right now into digging this as well but I can try to see if any fix helps my case. This is on Ubuntu 16.04 based system as well, with an elderly Intel HD 4000, at least judging by the Xorg.log.

**Comment by valff** [comment]
@jwilm Hey. I made some testing, but didn't figure out the issue. Firstly with debug build the problem is solving. I have an overclocked 4.2ghz cpu, so I guess it is a timing issue, which appears on fast enough cpus.

Also commenting out `terminal.needs_draw()` also helps, but I believe it is not the root of the issue. I made a debug logging, and everything looks fine, `Awakened` event is triggered correctly, and causes redraw afterwards.

Looks like the `terminal.draw()` renders a stale state. How can it be?

**Comment by divoxx** [comment]
This issue just started happening to me for the first time. I've been using alacritty as my main terminal for about 6 months and haven't seen the problem before, until now.

It also seemed to start happening without any version change or rebuild, which is very weird and no library that alacritty seems to depend on has been updated it.

**Comment by divoxx** [comment]
The problem seems to only happen with transparency enabled.

**Comment by divoxx** [comment]
This issue just started happening to me for the first time. I've been using alacritty as my main terminal for about 6 months and haven't seen the problem before, until now.

I was able to narrow down the context in which the problem happen, at least for me. It only happens when using a compositor (happens with both mutter/gnome and compton for me), and when there are multiple Alacritty windows open. It does not happen with composition and just one window.

**Comment by chrisduerr** [comment]
Might be related to #1227.

**Comment by dominik-zeglen** [comment]
It's happening on my i3-gaps + compton machine too. It seems that switching compton backend from `glx` to `xrender` helps.

**Comment by DrRibosome** [comment]
same issue:

- running on ubuntu 16.04
- built from hash `512fc6109182523c50e55ca8729b056442f36823`
- happens all the time, even outside of tmux, etc
- on startup, alacritty terminal background is the same as the window behind* the terminal, until I press a key and it redraws
- issue seems to happen once every 10 key strokes or so (pressing enter or control-c)
- clicking on the screen or typing a character seems to cause a redraw

**Comment by chrisduerr** [comment]
I have a hunch that this issue might be solved with #1518, since another similar issue was fixed with it. I've added this issue to the issues which will be automatically closed once that PR is merged.

If anyone is able to reproduce this issue with #1518, please let me know and I'll remove the reference to this issue.

**Comment by petobens** [comment]
Hi @chrisduerr I tried #1518 with #1403 and I'm still seeing the issue (randomly). No idea what's going on (is it related to the GPU?). This turns Alacritty a bit unusable :( (the only apparent way to force a redraw in my case by opening a new window and send the current (hanged) alacritty window to the back).

@dominik-zeglen the compton change fixes the issue for you?

**Comment by chrisduerr** [comment]
Huh, I was never able to reproduce this. But thanks for letting me know that #1518 does not fix this. I'll remove the auto-close message from the comments.

**Comment by atopia** [comment]
Not sure if this is the same issue, but on Arch Linux/X11/awesome, current master does not draw characters until a key press, mouse click or resize. Background color + opacity is drawn fine, but shell prompt only shows up after any of the three. I'll be offline until next Saturday but can help debug this next weekend. I don't recall the same issue on 0.2.0, but I might just not have noticed because the tiling modes cause a resize at startup.

**Comment by loewenheim** [comment]
I’m experiencing the same issue with i3-gaps and compton. Switching compton to xrender works, but feels a bit sluggish in general.

**Comment by chrisduerr** [comment]
I'm using i3-gaps and compton myself and I've never run into this issue. Which GPU are you using @loewenheim?

**Comment by loewenheim** [comment]
Here’s the output of `lspci`:
```
00:02.0 VGA compatible controller [0300]: Intel Corporation Skylake GT2 [HD Graphics 520] [8086:1916] (rev 07) (prog-if 00 [VG  controller])
	Subsystem: Lenovo Device [17aa:5053]
	Flags: bus master, fast devsel, latency 0, IRQ 126
	Memory at f0000000 (64-bit, non-prefetchable) [size=16M]
	Memory at e0000000 (64-bit, prefetchable) [size=256M]
	I/O ports at e000 [size=64]
	[virtual] Expansion ROM at 000c0000 [disabled] [size=128K]
	Capabilities: <access denied>
	Kernel driver in use: i915
	Kernel modules: i915
```

**Comment by Spencer-H** [comment]
I'm using i3-gaps with compton on version 0.3.2, this issue appeared when reflow support was added.
Alacritty will stop redrawing until I mouse over it or press a key if I have it highlighted.

**Comment by askobara** [comment]
I've faced same problem with Gnome on X11, but when I switch my DE to awesomeWM it's disappearing

**Comment by cj360** [comment]
Also seem to have this issue (I think anyways) I noticed long stretches of code like building a fork of aosp for Android or just playing music with mpd + ncmpcpp the terminal & visualizer will appear to hang but keyboard input or mouse movement will have it immediately redraw and catch up. This is on Arch Linux with Nvidia's vulkan beta drivers 418.52.14 on xorg-server 1.20.5-1.

**Comment by gbalke** [comment]
I have the same issue on Ubuntu 18.04 running i3 (Intel graphics accelerator). Generally occurs when I'm doing anything that prints lines continuously although I just noticed it with a git commit (froze half way through the print).

**Comment by kchibisov** [comment]
@gbalke you're observing this one #2217 .

**Comment by mooskagh** [comment]
Happens to me too. Usually I blamed ssh connection for that, but today it happened twice when running something locally. I don't use tmux or any other terminal multiplexer.

Happens much more frequently on my laptop, where I have ubuntu 14.04.6 LTS with Alacritty v0.2.9, but also happens at my workstation/desktop, where I run arch linux and have 0.3.3-2 installed.

Always happens when long (but not extremely long) output happening and scrolling the screen (like building something or `pacman -Syyuu`), but maybe it's just because I happen not to press keys while it's happening. (when I type, maybe it would happen too, but it's easy to be unnoticed).

**Comment by fcharras** [comment]
Happens every 30 seconds or so on my laptop, using i3 with Xcompmgr.

**Comment by mderrien29** [comment]
Getting this issue too with things like the npm package https://www.npmjs.com/package/concurrently or nvim.

Maybe multithread related ?

**Comment by sagis-tikal** [comment]
> 
> 
> Getting this issue on 0.3.3 too, with things like the npm package https://www.npmjs.com/package/concurrently or nvim being the biggest culprits.
> 
> Maybe multithread related ?

I wouldn't imagine it's a part of it - to me it also happens when I run the command 'watch -n0.1 date' or tail -f /some/file 
P.S. I'm using Manjaro Linux with i3wm under Virtual Box.

**Comment by mooskagh** [comment]
I can also confirm that it happens during streaming console output (`cat filename` or `git pull` or `make` to build something), it's not specific to being full-screen (like vim or tmux) or running multithread in some way.

**Comment by chrisduerr** [comment]
The issue you're looking for is https://github.com/jwilm/alacritty/issues/2217. Please don't spam either of them with +1's, since that will likely just burry useful information. If there's additional information required, I'll let you know, thanks.

**Comment by andys8** [comment]
I'm looking forward to glutin eventloop 2.0 which is actively worked on and will eventually fix the issue. But until then it drives me nuts. It happens a lot and I constantly press random keys, because I'm afraid the terminal isn't updating.

Is there a workaround to avoid having to switch to a different terminal emulator? E.g firing events constantly, disable optimization or to use an older version of alacritty?

**Comment by chrisduerr** [comment]
It would be possible to use an older version of Alacritty until the issue is resolved.

There are also some patches in the archlinux repository which downgrade the glutin version, applying those might solve the issue though their focus seems to be on resolving transparency problems.

I would also expect the evenloop 2.0 pull request to be usable soon on Linux. My plan is to resolve the remaining issues in Alacritty within this week, however that does not mean that there won't be any remaining issues upstream. However on macOS this won't help at all.

So those are the main options for now. The next Alacritty will definitely release with this fixed, hopefully relatively soon once the upstream macOS issues are resolved.

**Comment by inglor** [comment]
I'm using the pr2438 patched on top of 5cf77bf250712bf2630322f92c1025baa364922e (master) without issues (so far) at archlinux (locally build).

**Comment by logankaser** [comment]
I have this bug with alacritty 0.3.3 (06e52a6) on Ubuntu 14.04 (I know its old, work is stuck on it)

**Comment by chrisduerr** [comment]
I think I can reproduce this issue with the evlp2 branch. So this needs more investigation.

**Comment by andersevenrud** [comment]
I immediately updated now that I saw #2438 was merged. This seems to have solved all of the issues I had relating to this (and so much more) :heart_eyes:

**Comment by leosunmo** [comment]
This is great stuff! I would greatly appreciate a new release with this fix in it as I can finally recommend Alacritty for colleagues and projects!

**Comment by atopia** [comment]
As of 729eef0c, on my system (Arch Linux, X11, awesome WM ), a visual bell stays "on" until some interaction with the window is performed (such as moving the mouse):
`$ sleep 1; echo '\a'`

**Comment by sagis-tikal** [comment]
I've tested https://github.com/jwilm/alacritty/commit/729eef0c933831bccfeac6a355bdb410787fbe5f as well, seems like the issue was solved in it's entirety, no bell issues too.

**Comment by originalsouth** [comment]
> As of [729eef0](https://github.com/jwilm/alacritty/commit/729eef0c933831bccfeac6a355bdb410787fbe5f), on my system (Arch Linux, X11, awesome WM ), a visual bell stays "on" until some interaction with the window is performed (such as moving the mouse):
> `$ sleep 1; echo '\a'`

Can confirm this behavior for post 729eef0 on archlinux.

**Comment by ratorx** [comment]
I tried alacritty 0.3.3 and a75e9493449d90b54a140b76f6a96b684784a059 (Arch Linux; i3-gaps) and the issue persisted for me. I also tried termite and ran into the issue there as well. However, I can't reliably reproduce it (I just spam commands into the terminal until it eventually happens. The most reliable trigger seems to be trying to run commands with slightly delayed output such as `sudo`).

Also, I can't reproduce the issue with compton disabled (in either version). Running compton with `vsync = false` also seems to fix it for me.

**Comment by givensnickel** [comment]
If it's any help, I have found that with multiple windows open they all freeze at the same time and a keypress in any of them will unfreeze them all. 

I'm on Arch, with xfce for both desktop and wm.

**Comment by Dzordzu** [comment]
Note that as of today, arch package `alacritty-git` is out of date (v 0.3.3). This version still suffers from this issue. 
What's more (as it has been already said) [729eef0](https://github.com/jwilm/alacritty/commit/729eef0c933831bccfeac6a355bdb410787fbe5f) solved an issue. So the issue should be closed

**Comment by chrisduerr** [comment]
@Dzordzu That's a separate issue.

**Comment by FFY00** [comment]
@Dzordzu the package is not outdated. In fact, there's no package, there is a PKGBUILD (package build script). The PKGBUILD will fetch and build the last commit. There's nothing outdated.

**Comment by Dzordzu** [comment]
Fact - this is PKGBUILD script. But it doesn't matter, most of the community will name it "package".

And in the theory `alacritty-git` SHOULD do this (or build alacritty by tag name). In practise there is [a little change](https://aur.archlinux.org/packages/alacritty-git/#comment-691971)

**Comment by FFY00** [comment]
> Fact - this is PKGBUILD script. But it doesn't matter, most of the community will name it "package".

Yes, I just clarified it to explain the seccond part.

> And in the theory alacritty-git SHOULD do this (or build alacritty by tag name). In practise there is a little change

What little change? That comment is not related at all. The current PKGBUILD is fetching the last commit. Everything works as expected. I don't understand what you are trying to argue here.

**Comment by quininer** [comment]
@Dzordzu  I don't think git package **must** follow tag version, if you want, you can use [community package](https://www.archlinux.org/packages/community/x86_64/alacritty/).

**Comment by Dzordzu** [comment]
My bad. The problem with package versioning on my machine (that lasted for almost for a half a year from now) was on my side. I'm sorry for interrupting.

**Comment by chrisduerr** [comment]
I'd be curious if anyone was ever able to reproduce this with Wayland. It seems like this issue is exclusive to X11?

**Comment by jtroo** [comment]
Just want to add something that may be useful.

I was using the Debian distribution which was 0.2.9 and was reproducing the issue in X11 - Debian 10 + xfce. Tested in Weston (Wayland) in otherwise the same environment and the issue does not occur.

However, tried compiling master ( f48c43f ) and the issue no longer occurred in X11.

**Comment by FergusFettes** [comment]
Also having this issue on Ubuntu 20.04, running regolith (i3). Tried both building from source and installing with cargo to no avail.

However on the same computer it works when using the standard ubuntu gnome desktop. So seems in my case to be due to i3. I'll dig further and report back if I find the cause.

**Comment by leosunmo** [comment]
@FergusFettes Try disabling Compton.
```
killall compton
```

**Comment by fcharras** [comment]
Compton has a fork known has picom, switching to picom fixed the bug for me.

https://wiki.archlinux.org/index.php/Picom
https://github.com/yshui/picom

**Comment by blacksails** [comment]
I get this when building from source (master v0.8.0-dev) on ubuntu 20.10 GNOME on x11. Installing from apt (v0.6.0) does not seem to have the issue.

**Comment by marcus13345** [comment]
@chrisduerr I am having this issue on the latest release for macos. macos version: 12.1

**Comment by chrisduerr** [comment]
Exactly as described? I don't think I've ever seen this reported for macOS.

**Comment by tqbl** [comment]
@chrisduerr I'm also having an issue on macOS. Note that the output also updates if I wait several seconds.

The macOS version is 12.2, but I also had issues with Big Sur. The issue occurs with or without tmux. It also occurs whether I use fish or bash, so I don't think it's shell-specific.

**Comment by blacksails** [comment]
> @chrisduerr I'm also having an issue on macOS. Note that the output also updates if I wait several seconds.

I think you should report a separate issue. FYI I am running on macOS 12.1 with no issues. I only saw this issue on linux and what you describe doesn't sound like the same thing (waiting for output doesn't help). Also the fact that you have problems with firefox as well, could mean that your issue is not related to alacritty at all.

**Comment by marcus13345** [comment]
@chrisduerr yes i am having the issue as described. The output updates after a few seconds of inactivity _or_ with new input I'll happily make a new issue though

**Comment by tqbl** [comment]
> @chrisduerr yes i am having the issue as described. The output updates after a few seconds of inactivity _or_ with new input I'll happily make a new issue though

See #5883. Our issue is different from this one since it's severe lag as opposed to indefinite hanging.

**Comment by QuocAnhVu** [comment]
+1. I also have this bug for my system. At first, I thought it was tmux or vim but it is happening even outside of these programs. It is seemingly random and hard to trigger reliably. It also seems to come in waves.

Fedora 35
x11 + GDM/Gnome 3
Nvidia GPU + Nouveau drivers
Version: master ([dbccd7e30f40d9a7485c7537b415473ffc09b3d3](https://github.com/alacritty/alacritty/commit/dbccd7e30f40d9a7485c7537b415473ffc09b3d3))

**Comment by xulongwu4** [comment]
I have this issue on Solus Budgie with two laptops: one with intel CPU + Nvidia GPU, and the other one with AMD 5900HS + Nvidia GPU. They both show this issue when the laptop is connected to external monitors with the graphics rendered by nvidia GPU.

However, the issue seems to disappear when I switch to KDE or MATE desktop. I don't remember seeing people on KDE/MATE reporting this issue. If that is the case, it might be related to specific display managers/window managers rather than a general X11 issue.

**Comment by meeshaa** [comment]
started having this issue today on arch-x11-gnome for no apparent reason — i haven't even updated any package in the last few days

**Comment by QuocAnhVu** [comment]
I have some reason to believe that this is not an Alacritty bug. I noticed this same behavior when using Visual Studio Code (installed with flatpak).

**Comment by NiklasBeierl** [comment]
@meeshaa Same here.

**Comment by ajukraine** [comment]
I experience the same problem on my MacBook Pro 16. Issue also reproducible on Kitty and WezTerm terminals. But doesn't appear in iTerm2 (nor built-in Terminal.app).

It actually drives me nuts. I use terminal for code editing (via Vim/Neovim/tmux), because I want distraction free environment with fast response. But can't achieve that with terminal emulators. I have feeling the root cause is actually my system (hardware and OS). But how come iTerm2 avoids problem, and other modern GPU-accelerated terminals don't?...

**Comment by chrisduerr** [comment]
Alacritty master recently introduced a GLES2 renderer, I'd be curious if any of the people that can reproduce this want to test out opting into the slower renderer to see if that fixes it.

All you have to change is this line, which performs detection automatically:
https://github.com/alacritty/alacritty/blob/master/alacritty/src/renderer/mod.rs#L92

**Comment by QuocAnhVu** [comment]
> Alacritty master recently introduced a GLES2 renderer, I'd be curious if any of the people that can reproduce this want to test out opting into the slower renderer to see if that fixes it.
> 
> All you have to change is this line, which performs detection automatically: https://github.com/alacritty/alacritty/blob/master/alacritty/src/renderer/mod.rs#L92

I just tried forcing it both ways and it doesn't make a difference.

**Comment by NiklasBeierl** [comment]
I just found something that might actually help to uncover the underlying issue here. Hear me out, its a bit of a long story:

I have an i7-1065G7 CPU with Iris graphics inside a Dell XPS 13 7390 2-in-1.

I had a bunch of problems with anything that uses gpu acceleration over the last couple of weeks but a combination of: 
- Not sufficiently annoying, just puts extra load on my CPU
- Not getting anywhere with the usual tricks

... prevented me from resolving it. 

However, this weekend I switched to GNOME because I just go too frustrated with certain problems (not related to this one) in KDE Plasma. 

GNOME resolved my problems but introduced two others:
- This alacritty problem (Only in Xorg)
- gnome-shell made my CPU go apeshit from just moving the mouse around in idle (30-60% usage) (Wayland and Xorg)

[As described in my arch linux forum post](https://bbs.archlinux.org/viewtopic.php?id=277526): After two painfull afternoons of googling and poking around it turned out that gnome-shell fell back onto rendering on the CPU using `llvmpipe` because `mesa` was loading the wrong driver for my GPU!

I discovered this by checking `journalctl -b /us/bin/gnome-shell` and seeing warnings like these:
```
MESA-LOADER: failed to open i965: /usr/lib/dri/i965_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/dri, suffix _dri)
failed to load driver: i965
kmsro: driver missing
Failed to initialize accelerated iGPU/dGPU framebuffer sharing: Not hardware accelerated
libEGL warning: MESA-LOADER: failed to open i965: /usr/lib/dri/i965_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/dri, suffix _dri)
libEGL warning: MESA-LOADER: failed to open i965: /usr/lib/dri/i965_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/dri, suffix _dri)
```

MESA even attempting to load `i965` is odd, since my GPU needs the `iris` driver! Not `i965`!
So I decided to take a proper look at this, ended up setting: `MESA_LOADER_DRIVER_OVERRIDE=iris` in my `/etc/environment` and BOOM: 
- CPU load Problems gone

AND:

- This alacritty on XORG problem also gone

And this might also be the reason while it randomly affects certain people but is unreproducible for the maintainers of alacritty. 
MESA loading the wrong GPU driver can go unnoticed for a loooong time (in my case: weeks) since most applications will just fall back to tormenting your CPU. My Plasma was probably running on the CPU for weeks without me noticing!

**Comment by kchibisov** [comment]
Could someone try #6175. At least it fixes it for me in vm, since I never had it with anything else.

**Comment by spaddex** [comment]
@kchibisov, I'm running X11 with tmux and experience the issue mentioned here and in https://github.com/alacritty/alacritty/issues/4362. 
Have not experienced the issue since switching over to https://github.com/alacritty/alacritty/pull/6175 (been running it for ~3 days)

**Comment by acook** [comment]
Can confirm this happens with the latest released version of `0.11.0` but no longer is reproducible with the latest from git `0.12.0-dev (2d27fff7)`!

**Comment by nobrowser** [comment]
Still happens to me with 0.12.2 , on pop-os (based on Ubuntu) 22.04 LTS. I haven't seen it except with tmux as the shell, though I haven't tried very hard. With tmux, every couple of minutes :(

**Comment by charlesdaniels** [comment]
This is still happening for my on 0.12.2 on pop-os 22.04 as well.

```
$ alacritty --version
alacritty 0.12.2
$ uname -a
Linux eos 6.5.4-76060504-generic #202309191142~1695998943~22.04~070916d SMP PREEMPT_DYNAMIC Fri S x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/os-release
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
```

**Comment by borjasanlei** [comment]
Same here! This issue should be reopen or at least considered:
```bash
$ alacritty --version
alacritty 0.13.0-dev (a58fb39b)

$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

**Comment by trungda** [comment]
Same here! 

```
alacritty --version
alacritty` 0.14.0-dev (caf8c7fc)

cat /etc/os-release
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
```

**Comment by akyoto** [comment]
This issue comes up on Google search, just adding that this happens on my machine as well:

```
~ alacritty --version
alacritty 0.13.1 (fe2a3c56)
```

```
OS: Arch Linux x86_64
DE: GNOME 45.3
GPU: NVIDIA GeForce GTX 1070
```

**Comment by acook** [comment]
It seems like the maintainers aren't looking at this thread, it might be time to create a new issue and just link to this one.

**Comment by kchibisov** [comment]
We're aware of issue with nvidia/llvmpipe on X11, which is likely not our issue, but just how x11 works. there's #7251.

**Comment by kchibisov** [comment]
That's completely irrelevant whether they worked or not. The issue is that one frame on X11 is not being presented to the display for whatever reason when we draw and submit it, so for now the only viable solution would be to draw e.g. 3 times last redraw just to ensure that the frame goes to display.

X11 is just not great when it comes to hardware acceleration, because it was never designed with it in mind, so things break like that is not surprising. Keep in mind that contiguous redraw generally works, because it's contiguous. For now only nvidia and llvmpipe seems to be broken, and intel using legacy xf86-video-intel(shouldn't be used anyway).

**Comment by Inky-developer** [comment]
I have this issue on my laptop, which uses a AMD Ryzen 7 4700U with Radeon Graphics for hardware acceleration

**Comment by xfnty** [comment]
For me the issue was caused by `mouse.hide_when_typing = true`.
I use Artix/Wayland/Sway. I also noticed that when the terminal hangs
after i typed basically any command or pressed enter in nano for example
I could not move the mouse.

**Comment by kchibisov** [comment]
>after i typed basically any command or pressed enter in nano for example
I could not move the mouse.

that sounds like a GPU hang to me, maybe you have something in dmesg when such thing happen?

**Comment by d-s-a** [comment]
> For me the issue was caused by `mouse.hide_when_typing = true`. I use Artix/Wayland/Sway. I also noticed that when the terminal hangs after i typed basically any command or pressed enter in nano for example I could not move the mouse.
> 
> So the solution for me was to disable the flag.
> 
> Alacritty version is `alacritty 0.13.1 (fe2a3c56)`, got it from pacman.

Turning `hide_when_typing` off did fix it for me.
